### DISCUSSION TITLE
Is CoPilot available on Visual Studio for Mac 2022?

### DISCUSSION THREAD

**Discussion by jamesstephenbrown** [community/community#18691] [Copilot Conversations] [answered]
I'm part of the CoPilot program. I've seen an article showing how to get it working on PC, but in my Prefs/Extensions I don't see any options for CoPilot in the installs. Is there a downloadable extension installer for Mac?

**Comment by elsowiny** [comment]
Try signing into VScode with your Github account then when signed in check the extensions for GitHub Copilot, that is what I did for my mac.

**Reply by VirajShah21** [reply]
Visual Studio, not Visual Studio Code

**Reply by mahab339** [reply]
Take suggestions from there and return to visual studio.

**Comment by jayendranarumugam** [comment]
I think its currently not available for VS MAC

**Reply by MATBMS** [reply]
I confirm : https://docs.github.com/en/copilot/getting-started-with-github-copilot/getting-started-with-github-copilot-in-visual-studio#prerequisites

**Reply by hindams** [reply]
Not true

<img width="834" alt="image" src="https://github.com/community/community/assets/58893932/95e416a3-8a26-499d-be3f-3b3dee91dfef">

**Reply by montoulieu** [reply]
Did you read the words in the blue banner at the bottom of your image?

**Reply by josealvarez97** [reply]
The banner is still there as of Sep 20, 2023 😞 

https://docs.github.com/en/copilot/getting-started-with-github-copilot#prerequisites-1

<img width="759" alt="image" src="https://github.com/community/community/assets/21250471/9b9798e4-09a6-4dcb-9f62-071a2cd3eabc">

**Comment by heyemjay22** [comment]
Prerequisites. To use GitHub Copilot in Visual Studio, you must have Visual Studio 2022 17.2 or later installed. For more information, see the Visual Studio IDE documentation. Note: GitHub Copilot is not currently available for use with Visual Studio for Mac[.](https://heyemjay.us/)

**Comment by HiPPFormula** [comment]
[Prerequisites](https://hippformula.us/). To use GitHub Copilot in Visual Studio, you must have Visual Studio 2022 17.2 or later installed. For more information, see the Visual Studio IDE documentation. Note: GitHub Copilot is not currently available for use with Visual Studio for Mac.

**Comment by mrfsrf** [comment]
any idea on when will this be supported on osx visual studio code

**Reply by ArcusVonSinus** [reply]
It already is (https://docs.github.com/en/copilot/getting-started-with-github-copilot/getting-started-with-github-copilot-in-visual-studio-code)

**Reply by callummarshall9** [reply]
That's Visual Studio code... Not Visual Studio for Mac...

**Reply by ArcusVonSinus** [reply]
Indeed, but the question was about vs code.

**Reply by mrfsrf** [reply]
> any idea on when will this be supported on **OSX** visual studio code

**Reply by mrfsrf** [reply]
for some reason, nightly version of copilot works on OsX in VSCode

**Comment by aquari93** [comment]
Hey, Are there any plans to port Copilot to VS for mac ?

**Reply by abzork** [reply]
Yep, have been waiting for it for months. Github please make it happen :D

**Reply by danieltharris** [reply]
VS for Mac 2022 is such a huge improvement over 2019, and seemed to get a lot of love from MS, so I hope that will continue and we get this soon. One option is to use VS Code for now as a workaround. I still use VS on Windows too, but VS Mac 22 is lightweight and lightning fast on an Mx Pro machine

**Reply by aquari93** [reply]
Indeed VS for Mac using the M series cpu is a powerhouse. It's a shame that Copilot is still not available on the Mac. I used this feature on Windows and is awesome

**Comment by JuanuMusic** [comment]
Are there plans for implementing this in VS for Mac or is it not on the roadmap?

**Reply by danieltharris** [reply]
I would hope so as they seem pretty serious about VS for Mac up until now - Would be great to get confirmation

**Comment by depDesai16** [comment]
Does anyone know when Github Copilot will be available for Visual Studio for Mac?

**Comment by TPitman-mindfiretechnology** [comment]
I would also like to add my desire for CP for VS for Mac. I just purchased a monthly subscription and now realize I can't use it in VS for Mac. I do not use VSC very often. Most of my projects are in VS for Mac.

**Comment by minitockers** [comment]
Like the other comments, this would be a gamechanger, and really hope this is being worked on!

**Comment by renanpaiva64** [comment]
Waiting. BTW, VS on Mac has a huge lack of plugins.

**Comment by Alive-hub** [comment]
C'mon guys, we need Copilot for VS on Mac!

**Comment by dev-fredericfox** [comment]
At least a timeline would be great...

**Comment by AAChartModel** [comment]
Damn it, We need GitHub Copilot for VS for Mac,  Even though I can use it on Jetbrains Rider for now.

**Comment by Caldas** [comment]
Totally necessary and still don't have a timeline

**Comment by softgandhi** [comment]
+1
I am also waiting for CoPilot on VS for Mac.

**Comment by mimoid-prog** [comment]
Bump. We need Copilot for VS for Mac!

**Comment by Dezzley** [comment]
Yeah guys, please provide a timeline at least.
Copilot for VS for Mac would be highly appreciated. Especially given that both are MS products, it's sorta weird that it hasn't been released yet.

**Comment by VladDrgm** [comment]
Copilot for VS Mac would be amazing! I would drop Rider in an instant! Anyone know where can we vote for this so that it gets attention from Microsoft / Github? :)

**Reply by jdp-1902** [reply]
Here and MS Dev Community: https://developercommunity.visualstudio.com/t/Add-support-for-Github-Copilot/10320037?q=copilot

**Comment by speedhawk21** [comment]
Visual Studio for Mac with copilot would be a great plugin to support!  In the meantime I'll just have to sick with VSCode...

**Comment by achimeno** [comment]
Where is Copilot for Visual Studio for Mac?

**Comment by gandensang** [comment]
hmmm copilot not support for visual studio mac...

**Comment by renanpaivamobi** [comment]
I'll cancel my signature, because today I only use mac

**Comment by softgandhi** [comment]
https://developercommunity.visualstudio.com/t/Add-support-for-Github-Copilot/10320037?q=copilot

Request people to vote for faster action.

**Comment by UnChallenged** [comment]
we need copilot for visual studio mac

**Comment by WilCantrellAtDs** [comment]
The best way of writing iOS apps with Xamarin is with a Mac. How difficult is getting Copilot ported to VS for macOS? I'll even take Visual Studio for Arm64 so I can at least get the experience in a VM.

**Comment by artemkoloskov-admin** [comment]
Badly in the need for Copilot  on VS for Mac, guys. Please and thank you. And please speak to us, it is frustrating to have no clue on wether and when it will be enabled and if we need to adopt other tools to use it.

**Comment by wolfdavo** [comment]
Bump. Can't believe this isn't already available. What's the hold up MS?

**Comment by DuckScapePhilip** [comment]
Lets get this added.

**Comment by onlyargon** [comment]
It's time to have it on vs for mac 🥲

**Comment by puebloDeLaMuerte** [comment]
Bump! Badly in need indeed!

**Comment by JulesThuillier** [comment]
Bump ! Our team is mostly using Mac and we don't want VS Code...

**Comment by cwj1234567** [comment]
plz give us copilot

**Comment by hindams** [comment]
Is there an unofficial way to side load it at least?

**Reply by winkmichael** [reply]
chat.openai.com in a browser ):

**Comment by WilCantrellAtDs** [comment]
I ended up getting a license for the Rider IDE because of this. Rider doesn't seem to be working well for my MAUI project. Hoping to see this in VS Mac soon.

**Reply by winkmichael** [reply]
Copilot X works with Rider?

**Reply by WilCantrellAtDs** [reply]
I have Copilot in Rider, but not the chat. JetBrains is releasing their own product which is supposed to be more dotnetcentric...I don't know if it's X or not.

**Comment by TomSoPolaris** [comment]
Plus one for CoPilot for VS for Mac

**Comment by foreverLoveWisdom** [comment]
Please make Copilot available for VS for Mac 🙏 🙏 🙏

**Comment by BurnOutTrader** [comment]
Let's get it happening 🙌

**Comment by floolean** [comment]
One more for the cause 🙋🏻‍♂️

**Comment by suemayaheldursi** [comment]
hi - will this ever be available - would be great thanks!

**Comment by BurnOutTrader** [comment]
I just bought JetBrains rider IDE and it is like visual studio and vs code combined, only using it for quant trading in c# so I cant comment on other languages or building ui but for .Net it is amazing and it supports copilot. All the bugs with vs code and vs intelisense and omni sharp on mac make the platforms really bad compared to Windows versions. Rider is really worth it if you're on a mac.

**Reply by artemkoloskov-admin** [reply]
"Just use Rider" is not a solution here.

**Reply by BurnOutTrader** [reply]
Haha I really only have a single use case for my IDE 😁

**Reply by artemkoloskov-admin** [reply]
fair enough

**Reply by foreverLoveWisdom** [reply]
- Thanks 🙏 for sharing your exp with Rider. Maybe all of us should move to Rider so they can take action 😅

**Reply by WilCantrellAtDs** [reply]
Copilot in Rider is so intrusive I have to turn it off a lot. Rider is a great product, but VS on Mac (even compared to VS on Windows) is a nicer, clean experience.

**Comment by paulman5** [comment]
Would be awesome to see a mac version of the github copilot chat for Visual Studio!

**Comment by petersteele111** [comment]
Kind of silly that Mac does not have copilot still. I have an enterprise account with VS Studio as well as a year paid up with copilot. I also pay for chatGPT pro as well. I can keep using the online variant, but would really like to see native support for it in vs code on Mac as well. I can use it on windows just fine, and vs code on Mac without issue. This has been requested for almost, if not, for over a year. What's the hold up here?

**Reply by winkmichael** [reply]
Exact same boat as you. I also pay for Copilot X, and occasionally use it on Windows, though it is still largely better to alt+tab to chatGPT4

**Comment by LeonardoX77** [comment]
what are the plans to make this real?

**Reply by foreverLoveWisdom** [reply]
* Nothing from them yet, you can see how many similar comments/request there are above 😅

**Comment by skay97** [comment]
Since the extension still hasn't been officially ported by Microsoft, you can do it manually like so: 
https://stackoverflow.com/a/76737023
CoPilot now works on my installation of visual studio in parallels.

**Comment by jdp-1902** [comment]
Microsoft Answered.
> With today’s announcement, we’re redirecting our resources and focus to enhance Visual Studio and VS Code, optimizing them for cross-platform development. No new framework, runtime, or language support will be added to Visual Studio for Mac.

See this:
[Visual Studio for Mac Retirement Announcement](https://devblogs.microsoft.com/visualstudio/visual-studio-for-mac-retirement-announcement/)

**Reply by jayendranarumugam** [reply]
Here goes the official announcement!

[![enter image description here][1]][1]


  [1]:https://i.stack.imgur.com/O7oan.png

**Reply by jamesstephenbrown** [reply]
Is there a reason why someone would use Jetbrains with Unity over setting up VSCode for using with Unity? Is the setup easier?

**Reply by Alyxdev-xyz** [reply]
Rider is fantastic, lots of tools for refactoring about anything from renaming class + file names to suggesting modern syntax on code blocks (_I was ignorant about so many new C# features_). I was already used to WebStorm and compared to VSCode it was night and day.

Also it is very well integrated with Unity: Console + debugger + play controls from Rider.

You should give it a try with the free month trial.

**Comment by eliashezron** [comment]
Well, quite a way to say good bye, Was just getting hooked to visual studio on mac 🥺

**Comment by swensonic** [comment]
For anyone that has been following this thread for months now, I don't think CoPilot will ever be available for Visual Studio for Mac. Microsoft just announced that they will be discontinuing Visual Studio for Mac in August of 2024.

https://devblogs.microsoft.com/visualstudio/visual-studio-for-mac-retirement-announcement/

Does anyone read this differently than me?

I have moved on from VS Mac to Jetbrains Rider but was still hoping for CoPilot support for VS Mac.

**Comment by spullara** [comment]
I would say this is a lost cause and that VS for Mac is dead.

**Comment by mbadaz** [comment]
VS Code mac supports copilot wonderfully. I'm wondering what's real issue or constraint (if any) in adding Copilot support in VS Mac?

**Reply by artemkoloskov-admin** [reply]
The constraint is that VS for Mac is on it's death bed

**Reply by josealvarez97** [reply]
Yeah, VS for Mac is going to be deprecated. It's best to switch now to VS Code to get used to it.

I used to enjoy using VS for Mac for C# in Unity. VS Code wasn't that good for Unity and C# in the past, but honestly since I switched recently I realized that VS Code support for C# (and Unity) has improved a lot.

I thought I would miss VS for Mac more, but so far so good.

**Reply by ayim** [reply, marked as answer]
That's correct - VS for Mac is going to be retired in August of 2024.
Highly recommend trying out C# Dev Kit for VSCode - let us know what you think.
https://devblogs.microsoft.com/visualstudio/visual-studio-for-mac-retirement-announcement/

**Reply by pablontiv** [reply]
This is a very odd option as those extensions are failing every now and then. I just got this one https://github.com/microsoft/vscode-dotnettools/issues/567 and there is no way to get it work again.
It's unbelievable that MS decided to retire a product without supporting their main language.
